cmd/osmosisd/cmd/root.go

```diff
@@ -477,6 +482,77 @@ func overwriteConfigTomlValues(serverCtx *server.Context) error {
 	return nil
 }
 
+// overwrites app.toml values. Returns error if app.toml does not exist
+//
+// Currently, overwrites arbitrage-min-gas-fee value in app.toml if it is set to 0.005. Similarly,
+// overwrites the given viper config value.
+// Silently handles and skips any error/panic due to write permission issues.
+// No-op otherwise.
+func overwriteAppTomlValues(serverCtx *server.Context) error {
+	// Get paths to config.toml and config parent directory
+	rootDir := serverCtx.Viper.GetString(tmcli.HomeFlag)
+
+	configParentDirPath := filepath.Join(rootDir, "config")
+	configFilePath := filepath.Join(configParentDirPath, "app.toml")
+
+	fileInfo, err := os.Stat(configFilePath)
+	if err != nil {
+		return fmt.Errorf("failed to read in %s: %w", configFilePath, err)
+	} else {
+		// app.toml exists
+
+		// Get setting
+		currentArbitrageMinGasFeeValue := serverCtx.Viper.Get(mempoolConfigName + "." + arbitrageMinGasFeeConfigName)
+
+		// .x format at 0.x format are both valid.
+		if currentArbitrageMinGasFeeValue == oldArbitrageMinGasFeeValue || currentArbitrageMinGasFeeValue == "0"+oldArbitrageMinGasFeeValue {
+			// Set new value in viper
+			serverCtx.Viper.Set(mempoolConfigName+"."+arbitrageMinGasFeeConfigName, newArbitrageMinGasFeeValue)
+
+			defer func() {
+				if err := recover(); err != nil {
+					fmt.Printf("failed to write to %s: %s\n", configFilePath, err)
+				}
+			}()
+
+			// Check if the file is writable
+			if fileInfo.Mode()&os.FileMode(0200) != 0 {
```

Comment:
I think this implies if the file is not writeable, then the defer func wont trigger so there will be no feedback for the operator right?

Reply:
They would just not have their config updated which IMO is fine. The intention is to silently ignore the update if their file isn't writeable

Reply:
Right, I just think they would want to still get a log print saying it wasn't overwritten due to  permission issues

Reply:
[33a0a44](https://github.com/osmosis-labs/osmosis/pull/7368/commits/33a0a443f43e773716baf60908cd3458401c4aef)